opentelemetry-sdk/src/opentelemetry/sdk/metrics/export/batcher.py

```diff
@@ -80,11 +80,4 @@ def process(self, record) -> None:
         else:
             aggregator.take_checkpoint()
 
-        if self.stateful:
```

Comment:
How come this isn't necessary anymore?

Reply:
had a misconception of what this statement did - I realize the original batcher (before views) was about right with the exception of handling aggregators when stateless, so I just changed it to copy the aggregator when stateless so duplicate calls to process() won't break anything, and AFAICT this works.

Reply:
The deep copy looks like it's for exporting the checkpointed value despite other possible 'record' s coming in at the same time right?

Reply:
For stateful the copy is stored for the next process call with the same record so it can keep the statefulness. I don't think the statement has any other effect since it wouldn't apply when stateless. My views changes gave it a new effect of copying the aggregator over so when another process call is made with a reference to the same aggregator as before the data won't be zerod out by another take_checkpoint(), so its relevant to stateless as well.

Reply:
Sorry I don't really understand. Where does your Views changes copy the aggregator?

Reply:
Definitely badly worded - my views changes passes around references instead of copies of ViewDatas (and aggregators), so when we get to process() we can't just blindly merge aggregators together since they may be the same aggregator.

However if we copy the referenced aggregator to a new aggregator, you will be able to just merge any future aggregators with the same key without any issues. This is the new effect of the statement.

I also added/expanded the tests to run multiple ticks() stateless and stateful to test.